File: Радькова/2583.py
# Definition for a binary tree node.
# class TreeNode:
#    def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthLargestLevelSum(self, root: Optional[TreeNode], k: int) -> int:
        Q = list([root])
        lvlsum = []

        while Q:
            qlen = len(Q)
            tmpsum = 0

            for i in range(qlen):
                node = Q.pop(0)
                tmpsum += node.val

                if node.left:
                    Q.append(node.left)
                if node.right:
                    Q.append(node.right)

            lvlsum.append(tmpsum)

        if len(lvlsum) < k:
            return -1
        else:
            lvlsum.sort(reverse=True)

            return lvlsum[k-1]

File: Ивашечкин/2583.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func kthLargestLevelSum(root *TreeNode, k int) int64 {
    height := 0
    sum := []int64{} 
    sum, height = bfs(root, height)
    if height < k {
        return int64(-1)
    }
    if k > len(sum){
        return -1
    }
    sort.Slice(sum, func(i, j int) bool {
        return sum[i] > sum[j]
    })
    return sum[k-1]
   
}


func bfs(node *TreeNode, height int) ([]int64, int) {
    sum := []int64{}
    if height >= len(sum) { //расширение слайса
        sum = append(sum, 0)
    }
    sum[height] += int64(node.Val) //складываем ноду в свою высоту
    neighbours := []*TreeNode{node} //массив соседей ноды
    for len(neighbours) > 0 {
        height++
        if height >= len(sum) { //расширение слайса
            sum = append(sum, 0)
        }
        nextNodes := make([]*TreeNode, 0, len(neighbours)) 
        for _, node := range neighbours {
            if node.Right != nil {
                nextNodes = append(nextNodes, node.Right)
				sum[height] += int64(node.Right.Val)
            }
            if node.Left != nil {
				nextNodes = append(nextNodes, node.Left)
				sum[height] += int64(node.Left.Val)
			}
        }
        neighbours = nextNodes // расширяем масссив соседей
		
    }
    return sum, height
}